src/likelihoods.jl

```diff
@@ -0,0 +1,43 @@
+"""
+    Likelihood
+
+An abstract type for likelihoods. Likelihoods are used to model the uncertainities associated
+with the data.
+"""
+abstract type Likelihood end
+
+"""
+    GaussianLikelihood(σ²)
+
+Gaussian likelihood with `σ²` variance. This is to be used if we assume that the uncertainity 
+associated with the data follows a Gaussian distribution.
+
+```math
+    p(y|f) = Normal(y | f, σ²)
+```
+On calling, this would return a normal distribution with mean `f` and variance σ².
+"""
+struct GaussianLikelihood{T<:Real} <: Likelihood
+    σ²::Vector{T}
+    function GaussianLikelihood(σ²::T) where {T<:Real}
+        new{typeof(σ²)}([σ²])
+    end
```

Comment:
```suggestion
```

I don't think that we need this.

Reply:
Flux's `params()` seems to fail if the fields are not a vector. I found this kind of a [work around](https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/blob/master/src/basekernels/constant.jl#L51-L56) in `KernelFunctions.jl`

Reply:
Oh, sorry, yeah, you're totally right.

Reply:
That would be amazing if one could use `Ref`... Do we know if there is any plan on this?